File: In_class_Exercise_2/Laptop.cpp
#include <iostream>
using namespace std;

class Laptop {
    private:
        string brand;
        string model;
        double price;
    public:
        Laptop(string b, string m, double p) : brand(b), model(m), price(p) {}

        Laptop() : Laptop("A", "B", 999.0) {
            cout << "-> Default constructor triggered chaining." << endl;
        }

        string getBrand() const { return brand; }
        string getModel() const { return model; }   
        double getPrice() const { return price; }
};

int main() {
    cout << "--- Creating object 1 (Default) ---" << endl;
    Laptop lap1;
    cout << "Brand: " << lap1.getBrand() << ", Model: " << lap1.getModel() << ", Price: $" << lap1.getPrice() << endl;

    cout << "\n--- Creating object 2 (Parameterized) ---" << endl;
    Laptop lap2("MacBook", "Pro M3", 2000.00);
    cout << "Brand: " << lap2.getBrand() << ", Model: " << lap2.getModel() << ", Price: $" << lap2.getPrice() << endl;

    return 0;
}

File: In_class_Exercise_2/Laptop.java
// package In_class_Exercise_2;

public class Laptop {
    private String brand;
    private String model;
    private double price;

    public Laptop(String brand, String model, double price) {
        this.brand = brand;
        this.model = model;
        this.price = price;
    }

    public Laptop() {
        this("Unknown Brand", "Unknown Model", 0.0); 
        System.out.println("-> Default constructor triggered chaining.");
    }

    // Main function
    public static void main(String[] args) {
        System.out.println("--- Creating object 1 (Default) ---");
        Laptop lap1 = new Laptop(); 
        System.out.println("Brand: " + lap1.brand + ", Model: " + lap1.model + ", Price: $" + lap1.price);

        System.out.println("\n--- Creating object 2 (Parameterized) ---");
        Laptop lap2 = new Laptop("Dell", "XPS 15", 1500.00);
        System.out.println("Brand: " + lap2.brand + ", Model: " + lap2.model + ", Price: $" + lap2.price);
    }
}